Lib/asyncio/selector_events.py

```diff
@@ -1135,3 +1140,120 @@ def _sendto_ready(self):
             self._loop._remove_writer(self._sock_fd)
             if self._closing:
                 self._call_connection_lost(None)
+
+
+class _SelectorSocketDatagramTransport(_SelectorTransport):
+
+    _buffer_factory = collections.deque
+
+    def __init__(self, loop, sock, protocol, address, waiter=None, extra=None):
+        super().__init__(loop, sock, protocol, extra)
+        self._address = address
+        self._closing_waiter = None
+        self._loop.call_soon(self._protocol.connection_made, self)
+        # only start reading when connection_made() has been called
+        self._loop.call_soon(self._loop._add_reader, self._sock_fd,
+                             self._read_ready)
+        if waiter is not None:
+            # only wake up the waiter when connection_made() has been called
+            self._loop.call_soon(futures._set_result_unless_cancelled,
+                                 waiter, None)
+
+    def _read_ready(self):
+        if self._conn_lost:
+            return
+        try:
+            data = self._sock.recv(self.max_size)
+        except (BlockingIOError, InterruptedError):
+            pass
+        except OSError as exc:
+            self._protocol.error_received(exc)
+        except Exception as exc:
+            self._fatal_error(exc, 'Fatal read error on datagram transport')
+        else:
+            self._protocol.datagram_received(data)
+
+    def send(self, data):
+        if not isinstance(data, (bytes, bytearray, memoryview)):
+            raise TypeError('data argument must be a bytes-like object, '
+                            'not %r' % type(data).__name__)
+        if not data:
+            return
```

Comment:
Are you sure about the semantics? Can't you use send ("") as a cheap ping for example?

I suggest to remove this test.